### DISCUSSION TITLE
Domain Discovery: Same SSO platform across multiple domains

### DISCUSSION THREAD

**Discussion by pavja2** [zitadel/zitadel#6300] [Q&A]
Trying to work through the domain discovery example from https://zitadel.com/docs/guides/solution-scenarios/domain-discovery with a slight modification.

![image](https://github.com/zitadel/zitadel/assets/1340709/55020d61-b785-4133-b830-d1ad8a4bdaa6)

What happens if Company `Alpha` enables Google Workspace as a third party IDP and one of their customers clicks the "Sign in With Google" button on `login.mycompany.com`?

Right now it looks like that creates an account for the user in the `CIAM` organization (e.g. `alice@alpha.com` will get an account in `CIAM`) but would I'd want to have happen is if `alice@alpha.com` signs in with Google Workspace she gets mapped to the `Alpha` organization instead.

Ideally what I'd want to have happen is:

- If someone Signs in with Google and their email address matches an organization domain for an organization that has enabled Sign in with Google - create an account for them in that Organization.
- Otherwise create an account for them in the CIAM Organization.

Is this possible to implement / what would I change from the tutorial to do so? Alternatively, is there a reason this is a bad way to architect domain discovery?

**Comment by mffap** [comment]
> If someone Signs in with Google and their email address matches an organization domain for an organization that has enabled Sign in with Google - create an account for them in that Organization.

The assumption in the guide is that the users enter their email address in the first screen (enter username). With that ZITADEL will apply the correct policies based on the organization's configuration.

If alice@alpha.com enters her email, then the next step in the login will be based on the policies provided by Alpha Organizations (ie. login with Google Workspace).

> Otherwise create an account for them in the CIAM Organization.

You can define a default organization, if none other is matched.

**Reply by pavja2** [reply]
Thanks! That's super helpful. How does registration work in this scenario? 

If I disable Registration on the Instance, but leave it enabled on the default Organization, then if people put in an email that hasn't been registered at the instance level it gets rejected as "User could not be found" instead of redirecting them to the Default Organization page where they can choose an SSO provider.

I'd want people to either be directed to their Company's login/registration flow, or to the default login/registration page if their email doesn't match any other organization. However, I can't seem to figure out the second part of that process.
